gpytorch/variational/variational_strategy.py

```diff
@@ -105,52 +115,64 @@ def forward(self, x):
-                    eager_rhs[..., :left_tensors.size(-1)].detach()
-                ]
-                solves = [
-                    solve.detach(), solve[..., left_tensors.size(-1):].detach(),
-                    solve[..., :left_tensors.size(-1)].detach()
-                ]
-                if settings.skip_logdet_forward.on():
-                    eager_rhss.append(torch.cat([probe_vecs, left_tensors], -1))
-                    solves.append(torch.cat([probe_vec_solves, solve[..., :left_tensors.size(-1)]], -1))
-            induc_induc_covar = CachedCGLazyTensor(
-                induc_induc_covar, eager_rhss=eager_rhss, solves=solves, probe_vectors=probe_vecs,
-                probe_vector_norms=probe_vec_norms, probe_vector_solves=probe_vec_solves,
-                probe_vector_tmats=tmats,
-            )
-
-            # Cache the prior distribution, for faster training
-            if self.training:
-                prior_dist = MultivariateNormal(induc_mean, induc_induc_covar)
-                self._prior_distribution_memo = prior_dist
 
             # Compute predictive mean/covariance
             inv_products = induc_induc_covar.inv_matmul(induc_data_covar, left_tensors.transpose(-1, -2))
             predictive_mean = torch.add(test_mean, inv_products[..., 0, :])
             predictive_covar = RootLazyTensor(inv_products[..., 1:, :].transpose(-1, -2))
-
-            # Compute a diagonal correction for the covariance. It is the diagonal of the Schur complement
-            # K_{data,data} - K_{data,induc} K_{induc,induc}^{-1} K_{induc,data}
             if beta_features.diagonal_correction.on():
-                # interp_data_data_var = diag(K_{data,induc} K_{induc,induc}^{-1} K_{induc,data})
-                interp_data_data_var = induc_induc_covar.inv_quad(induc_data_covar, reduce_inv_quad=False)
+                interp_data_data_var, logdet = induc_induc_covar.inv_quad_logdet(
+                    induc_data_covar, logdet=(self.training), reduce_inv_quad=False
+                )
                 diag_correction = DiagLazyTensor((data_data_covar.diag() - interp_data_data_var).clamp(0, math.inf))
                 predictive_covar = PsdSumLazyTensor(predictive_covar, diag_correction)
 
+            # Save the logdet, mean_diff_inv_quad, prior distribution for the ELBO
+            if self.training:
+                self._memoize_cache["prior_distribution_memo"] = MultivariateNormal(induc_mean, induc_induc_covar)
+
             return MultivariateNormal(predictive_mean, predictive_covar)
 
     def __call__(self, x):
         if not self.variational_params_initialized.item():
-            self.variational_distribution.initialize_variational_distribution(self.prior_distribution)
+            prior_dist = self.prior_distribution
+            eval_prior_dist = torch.distributions.MultivariateNormal(
+                loc=prior_dist.mean,
+                covariance_matrix=prior_dist.covariance_matrix
+            )
```

Comment:
I like this as a way of dealing with the different initializations! Is there any way we can pull it out of `__call__` though? it would be nice if we didn't have to overwrite `__call__` in any of the subclasses.